alpha/apps/kaltura/lib/batch2/bcdl/kBusinessPreConvertDL.php

```diff
@@ -279,19 +285,30 @@ protected static function setIsDefaultThumb(thumbAsset $thumbAsset)
 		}
 	}
 
-	private static function generateThumbnail(asset $srcAsset, thumbParamsOutput $destThumbParamsOutput, &$errDescription, $rotate=null)
+	private static function generateThumbnail(asset $srcAsset, thumbParamsOutput $destThumbParamsOutput, &$errDescription, $rotate=null, $entryDataPath = null, $fileSync = null)
 	{
-		$srcSyncKey = $srcAsset->getSyncKey(flavorAsset::FILE_SYNC_FLAVOR_ASSET_SUB_TYPE_ASSET);
-		/* @var $fileSync FileSync */
-		list($fileSync, $local) = kFileSyncUtils::getReadyFileSyncForKey($srcSyncKey, true, false);
+		if (!$fileSync)
+		{
+			$srcSyncKey = $srcAsset->getSyncKey(flavorAsset::FILE_SYNC_FLAVOR_ASSET_SUB_TYPE_ASSET);
+			/* @var $fileSync FileSync */
+			list($fileSync, $local) = kFileSyncUtils::getReadyFileSyncForKey($srcSyncKey, true, false);
 
-		if(!$fileSync || $fileSync->getFileType() == FileSync::FILE_SYNC_FILE_TYPE_URL)
+			if(!$fileSync || $fileSync->getFileType() == FileSync::FILE_SYNC_FILE_TYPE_URL)
```

Comment:
All file sync we have today at periodic storage are of type URL, what will happen in this case?